File: EstruturasAlgoritmos/suma.c
#include "stdio.h"
#include "stdlib.h"


float suma(int argc, char** argv){
  float Result = 0.0;
  int i;
  puts(argv[1]);
  for (i=1; i < argc; i++) {
    Result = Result + atof(argv[i]);
    printf("--%f\n", Result);
  }
  printf("-%f\n", Result);
  return Result;
}

int main(int argc, char** argv) {
  float Result = suma(argc, argv);
  printf("El resultado de la suma es: %f\n", Result);
  return 0;
}

File: EstruturasAlgoritmos/suma.py
#!/usr/bin/python

import sys

def suma():
    """
    Esta funcion suma n numeros.
    """
    Arguments = sys.argv[1:]
    Result = 0.0
    for argument in Arguments:
        Result += float(argument)
    print "El resultado es: " + str(Result)

suma()
